### PR TITLE
Add $sort modifier tests

### PR DESCRIPTION
**Pull request by PatersonProjects**
This PR adds tests for the $sort update modifier operator, testing the proper function and error codes with different BSON types, as well as proper ordering and edge cases of that ordering. 

Ref: #37, #391

### CODE REVIEW

File: documentdb_tests/compatibility/tests/core/operator/update/modifiers/sort/test_update_sort_errors.py [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+        {
+            "update": collection.name,
+            "updates": [{"q": test_case.query, "u": test_case.update}],
+        },
+    )
+    assertResult(result, error_code=test_case.error_code, msg=test_case.msg)
+
+
+@pytest.mark.parametrize("test_case", pytest_params(ALL_SUCCESS_TESTS))
+def test_update_sort_edge_cases(collection, test_case):
```
Comment (eerxuan) [inline documentdb_tests/compatibility/tests/core/operator/update/modifiers/sort/test_update_sort_errors.py]: duplicate function name test_update_sort_edge_cases across two files.
Rename the one in test_update_sort_errors.py to match its actual content (e.g. test_update_sort_missing_each_and_containing_array)

File: documentdb_tests/compatibility/tests/core/operator/update/modifiers/sort/test_update_sort_core.py [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+    UpdateTestCase(
+        id="empty_each_document_sort_descending",
+        setup_docs=[{"_id": 1, "arr": [{"x": 1}, {"x": 3}, {"x": 2}]}],
+        query={"_id": 1},
+        update={"$push": {"arr": {"$each": [], "$sort": {"x": -1}}}},
+        expected=[{"_id": 1, "arr": [{"x": 3}, {"x": 2}, {"x": 1}]}],
+        msg="Empty $each with document sort descending should sort existing array",
+    ),
+    UpdateTestCase(
+        id="multi_key_sort",
```
Comment (eerxuan) [inline documentdb_tests/compatibility/tests/core/operator/update/modifiers/sort/test_update_sort_core.py:85]: Major — multi_key_sort is a false-positive: input is already pre-sorted

  test_update_sort_core.py:97-124 uses input [{a:1,b:3},{a:1,b:1},{a:2,b:2},{a:2,b:1}] with sort spec {"a": 1, "b": -1} and expects... the exact same order. Since stable sort preserves the input order when the sort
  already holds, a no-op implementation (or one that ignores the sort spec entirely) would also pass this test. Reorder the input so the multi-key sort actually has work to do — e.g.
  [{a:2,b:1},{a:1,b:1},{a:2,b:2},{a:1,b:3}] would force the operator to actually reorder and produce the documented result.

File: documentdb_tests/compatibility/tests/core/operator/update/modifiers/sort/test_update_sort_errors.py [hunk 3]
```diff
@@ -0,0 +1,187 @@
+"""Tests for $sort update modifier error cases.
+
+Covers: missing $each, $sort with incompatible operators ($addToSet, $pull,
+$pullAll, $pop), target not array, and containing array field path behavior.
+"""
+
```
Comment (eerxuan) [inline documentdb_tests/compatibility/tests/core/operator/update/modifiers/sort/test_update_sort_errors.py:6]: Minor — test_update_sort_errors.py houses success-path tests

  MISSING_EACH_TESTS (line 752) and CONTAINING_ARRAY_FIELD_PATH_TESTS (line 809) both assert successful update results, not errors — but the file is named _errors.py and its docstring says "Covers… error cases." Either
  move these into test_update_sort_edge_cases.py (where similar non-error edge cases live) or rename the file.

File: documentdb_tests/compatibility/tests/core/operator/update/modifiers/sort/test_update_sort_errors.py [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+            {
+                "_id": 1,
+                "arr": [{"score": 3}, {"score": 1}, {"score": 2}],
+            }
+        ],
+        msg="Sort by containing array field path has no matching fields, order unchanged",
+    ),
+]
+
+PUSH_WITHOUT_EACH_LITERAL_TESTS: list[UpdateTestCase] = []
```
Comment (eerxuan) [inline documentdb_tests/compatibility/tests/core/operator/update/modifiers/sort/test_update_sort_errors.py]: Minor — dead PUSH_WITHOUT_EACH_LITERAL_TESTS = []

File: documentdb_tests/compatibility/tests/core/operator/update/modifiers/sort/test_update_sort_errors.py (hunk 1 above)
Comment (PatersonProjects) [reply documentdb_tests/compatibility/tests/core/operator/update/modifiers/sort/test_update_sort_errors.py]: Removed executor and moved tests to edge cases

File: documentdb_tests/compatibility/tests/core/operator/update/modifiers/sort/test_update_sort_core.py (hunk 2 above)
Comment (PatersonProjects) [reply documentdb_tests/compatibility/tests/core/operator/update/modifiers/sort/test_update_sort_core.py:85]: Rewrote input to properly trigger sorting, good catch.

File: documentdb_tests/compatibility/tests/core/operator/update/modifiers/sort/test_update_sort_errors.py (hunk 4 above)
Comment (PatersonProjects) [reply documentdb_tests/compatibility/tests/core/operator/update/modifiers/sort/test_update_sort_errors.py]: Removed

File: documentdb_tests/compatibility/tests/core/operator/update/modifiers/sort/test_update_sort_errors.py (hunk 3 above)
Comment (PatersonProjects) [reply documentdb_tests/compatibility/tests/core/operator/update/modifiers/sort/test_update_sort_errors.py:6]: Moved into edge cases

**Review by eerxuan — APPROVED** [review]
